test/lifecycle_module_test.dart

```diff
@@ -50,11 +52,11 @@ class TestLifecycleModule extends LifecycleModule {
   List<String> eventList;
   bool mockShouldUnload;
 
-  TestLifecycleModule()
+  TestLifecycleModule({String name})
       : managedDisposable = new Disposable(),
         managedStreamController = new StreamController<Null>(),
         managedStreamSubscription = new MockStreamSubscription(),
-        name = 'TestLifecycleModule' {
+        name = name ?? 'TestLifecycleModule' {
```

Comment:
@georgelesica-wf This sets a final String property on which overrides the base implementation for the name getter.